katharsis-core/src/main/java/io/katharsis/resource/registry/responseRepository/ResourceRepositoryAdapter.java

```diff
@@ -1,109 +1,189 @@
-			resources = ((QuerySpecResourceRepository) resourceRepository).findAll(ids,
-					toQuerySpec(queryAdapter, resourceInformation.getResourceClass()));
-		}
-		else {
-			resources = ((ResourceRepository) resourceRepository).findAll(ids, toQueryParams(queryAdapter));
-		}
-		return getResponse(resourceRepository, resources, new RequestSpec(queryAdapter));
-	}
-
-	public <S extends T> JsonApiResponse save(S entity, QueryAdapter queryAdapter) {
-		Object resource;
-		if (isAnnotated) {
-			resource = ((AnnotatedResourceRepositoryAdapter) resourceRepository).save(entity);
-		}
-		else if (resourceRepository instanceof QuerySpecResourceRepository) {
-			resource = ((QuerySpecResourceRepository) resourceRepository).save(entity);
-		}
-		else {
-			resource = ((ResourceRepository) resourceRepository).save(entity);
-		}
-		return getResponse(resourceRepository, resource, new RequestSpec(queryAdapter));
-	}
-
-	public JsonApiResponse delete(ID id, QueryAdapter queryAdapter) {
-		if (isAnnotated) {
-			((AnnotatedResourceRepositoryAdapter) resourceRepository).delete(id, queryAdapter);
-		}
-		else if (resourceRepository instanceof QuerySpecResourceRepository) {
-			((QuerySpecResourceRepository) resourceRepository).delete(id);
-		}
-		else {
-			((ResourceRepository) resourceRepository).delete(id);
-		}
-		return new JsonApiResponse();
-	}
-
-	public Object getResourceRepository() {
-		return resourceRepository;
-	}
-
-	@Override
-	protected Class<?> getResourceClass(Object repository) {
-		return resourceInformation.getResourceClass();
-	}
-}
+package io.katharsis.resource.registry.responseRepository;
+
+import java.io.Serializable;
+
+import io.katharsis.dispatcher.controller.HttpMethod;
+import io.katharsis.module.ModuleRegistry;
+import io.katharsis.queryspec.QuerySpecResourceRepository;
+import io.katharsis.queryspec.internal.QueryAdapter;
+import io.katharsis.repository.ResourceRepository;
+import io.katharsis.repository.annotated.AnnotatedResourceRepositoryAdapter;
+import io.katharsis.repository.filter.RepositoryFilterContext;
+import io.katharsis.request.repository.RepositoryRequestSpec;
+import io.katharsis.resource.information.ResourceInformation;
+import io.katharsis.response.JsonApiResponse;
+
+/**
+ * A repository adapter for resource repository.
+ */
+@SuppressWarnings("unchecked")
+public class ResourceRepositoryAdapter<T, I extends Serializable> extends ResponseRepository {
+
+	private final Object resourceRepository;
+
+	private final boolean isAnnotated;
+
+	public ResourceRepositoryAdapter(ResourceInformation resourceInformation, ModuleRegistry moduleRegistry,
+			Object resourceRepository) {
+		super(resourceInformation, moduleRegistry);
+		this.resourceRepository = resourceRepository;
+		this.isAnnotated = resourceRepository instanceof AnnotatedResourceRepositoryAdapter;
+	}
+
+	public JsonApiResponse findOne(I id, QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 23 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)